src/zsf_config.c

```diff
@@ -20,8 +19,15 @@ int _zsf_ini_handler(char *section, char *key, char *value, void *data_ptr) {
 
   errno = 0;
   if (!strcmp(section, "sealock")) {
-    config_ptr->num_locks = 1;
-    if (!strcmp(key, "computation_mode")) {
+    sealock_index_t lock_index = config_ptr->num_locks - 1;
+    assert(!*key || (*key && lock_index >= 0));
```

Comment:
The "*key &&" part is superfluous, it is already checked by the !*key.

Reply:
You are right, fixed it. The first term was added later.